bcda/client/bluebutton_test.go

```diff
@@ -328,35 +332,103 @@ func (s *BBRequestTestSuite) TearDownAllSuite() {
 	s.ts.Close()
 }
 
-func (s *BBTestSuite) TestAddRequestHeaders() {
-
-	bbServer := os.Getenv("BB_SERVER_LOCATION")
-
-	req, err := http.NewRequest("GET", bbServer, nil)
-	assert.Nil(s.T(), err)
-	reqID := uuid.NewRandom()
-	assert.Nil(s.T(), err)
-
-	params := url.Values{}
-	params.Set("_format", "application/fhir+json")
-
-	req.URL.RawQuery = params.Encode()
-	client.AddRequestHeaders(req, reqID, "543210", "A00234")
-
-	assert.Equal(s.T(), reqID.String(), req.Header.Get("BlueButton-OriginalQueryId"))
-	assert.Equal(s.T(), "1", req.Header.Get("BlueButton-OriginalQueryCounter"))
-
-	assert.Equal(s.T(), "", req.Header.Get("keep-alive"))
-	assert.Equal(s.T(), "https", req.Header.Get("X-Forwarded-Proto"))
-	assert.Equal(s.T(), "", req.Header.Get("X-Forwarded-Host"))
-
-	assert.Equal(s.T(), req.URL.String(), req.Header.Get("BlueButton-OriginalUrl"))
-	assert.Equal(s.T(), req.URL.RawQuery, req.Header.Get("BlueButton-OriginalQuery"))
-
-	assert.Equal(s.T(), "543210", req.Header.Get("BCDA-JOBID"))
-	assert.Equal(s.T(), "A00234", req.Header.Get("BCDA-CMSID"))
-	assert.Equal(s.T(), "mbi", req.Header.Get("IncludeIdentifiers"))
+func (s *BBRequestTestSuite) TestValidateRequestHeaders() {
+	tests := []struct {
+		name          string
+		funcUnderTest func(bbClient *client.BlueButtonClient, jobID, cmsID string) (string, error)
+	}{
+		{
+			"ExplanationOfBenefit",
+			func(bbClient *client.BlueButtonClient, jobID, cmsID string) (string, error) {
+				return s.bbClient.GetExplanationOfBenefit("patient1", jobID, cmsID, "", time.Now())
+			},
+		},
+		{
+			"GetPatient",
+			func(bbClient *client.BlueButtonClient, jobID, cmsID string) (string, error) {
+				return s.bbClient.GetPatient("patient2", jobID, cmsID, "", time.Now())
+			},
+		},
+		{
+			"GetCoverage",
+			func(bbClient *client.BlueButtonClient, jobID, cmsID string) (string, error) {
+				return s.bbClient.GetCoverage("beneID1", jobID, cmsID, "", time.Now())
+			},
+		},
+		{
+			"GetPatientByIdentifierHash",
+			func(bbClient *client.BlueButtonClient, jobID, cmsID string) (string, error) {
+				return s.bbClient.GetPatientByIdentifierHash("hashedIdentifier", "patientIdMode")
+			},
+		},
+	}
 
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(t *testing.T) {
```

Comment:
Maybe, in the future as part of BCDA-3398, we can mark this as `t.Parallel()`?

Reply:
Yeah that's a great idea. We'll definitely quite a bit of speed up by running package tests concurrently, and then we can add in parallelization within each file test.